Add a contributing guide that can be applied as a default to all repos (#6)
appwrite/.github: CONTRIBUTING.md

# Contributing

We would ❤️ for you to contribute to Appwrite and help make it better! We want contributing to Appwrite to be fun, enjoyable, and educational for anyone and everyone. All contributions are welcome, including issues, new docs as well as updates and tweaks, blog posts, workshops, and more.

## How to Start?

> If you are worried or don’t know where to start, check out our next section explaining what kind of help we could use and where can you get involved. You can reach out with questions on our [Discord](https://discord.gg/GSeTUeA) server. You can also submit an issue and a maintainer can guide you!

Reviewer: ```suggestion
If you are worried or don’t know where to start, check out our next section explaining what kind of help is needed and where you can get involved. You can reach out with questions on our [Discord](https://discord.gg/GSeTUeA) server. You can also submit an issue and a maintainer can guide you!
```

Reviewer: Awk

## Code of Conduct

Help us keep Appwrite open and inclusive. Please read and follow our [Code of Conduct](/CODE_OF_CONDUCT.md).

## Submit a Pull Request 🚀

Branch naming convention is as follows

`TYPE-ISSUE_ID-DESCRIPTION`

example:

```
doc-548-submit-a-pull-request-section-to-contribution-guide
```

When `TYPE` can be:

- **feat** - is a new feature
- **doc** - documentation only changes
- **cicd** - changes related to CI/CD system
- **fix** - a bug fix
- **refactor** - code change that neither fixes a bug nor adds a feature

**All PRs must include a commit message with the description of the changes made!**

> For the initial start, fork the project and use git clone command to download the repository to your computer. A standard procedure for working on an issue would be to:

Reviewer: ```suggestion
Start by forking the project and use the `git clone` command to download the repository to your computer. A standard procedure for working on an issue would be to:
```

1. `git pull`, before creating a new branch, pull the changes from upstream. Your master needs to be up to date.

```
$ git pull
```

> 2. Create new branch from `master` like: `doc-548-submit-a-pull-request-section-to-contribution-guide`

Reviewer: ```suggestion
2. Create a new branch from `master`. For example `doc-548-submit-a-pull-request-section-to-contribution-guide`
```

Author: Should we maybe say "default branch" since it might be main or master?

```
2. Create a new branch from the default branch. For example `doc-548-submit-a-pull-request-section-to-contribution-guide`
```

```
$ git checkout -b [name_of_your_new_branch]
```

3. Work - commit - repeat ( be sure to be in your branch )
4. Push changes to GitHub

```
$ git push origin [name_of_your_new_branch]
```

6. Submit your changes for review. If you go to your repository on GitHub, you'll see a `Compare & pull request` button. Click on that button.
7. Start a Pull Request (PR) by clicking on `Create pull request`. Make sure to update the PR description following the template provided.
8. Wait for a code review.
9. If you need to make changes based on feedback, make sure to re-request a review from your reviewer after you've made the necessary changes.

![Re-Request a Review](https://docs.github.com/assets/cb-4714/images/help/pull_requests/request-re-review.png)

10. After approval, your PR will be merged.

> 11. You can delete your branch after it has been merged.

Author: I made some tweaks here compared to:

https://github.com/appwrite/appwrite/blob/44ca7d9459dba9b21760e2934decbfd41f26c1ba/CONTRIBUTING.md?plain=1#L75-L81

Reviewer: But also, they'd be forking, right? This would delete their fork's branch.

Author: Yes, correct.

## Introducing New Features

We would 💖 you to contribute to Appwrite, but we would also like to make sure Appwrite is as great as possible and loyal to its vision and mission statement 🙏.

For us to find the right balance, please open an issue explaining your ideas before introducing a new pull request.

This will allow the Appwrite community to have sufficient discussion about the new feature value and how it fits in the product roadmap and vision.

> This is also important for the Appwrite lead developers to be able to give technical input and different emphasis regarding the feature design and architecture. Some bigger features might need to go through our [RFC process](https://github.com/appwrite/rfc).

Author: I'm not sure if we want this considering there's a "Submitting New Ideas" section down below too.

Reviewer: 🤔 We should avoid multiple ways to do the same thing.

Also maybe we can just say `... for Appwrite maintainers to discuss the feature's design and architecture to help the contribution fit in organically in the Appwrite codebase.`

I think this sounds more welcoming. I don't want to give the impression that our lead engineers will scrutinize the PR in detail.

### Blogging & Speaking

Blogging, speaking about, or creating tutorials about one of Appwrite’s many features. Mention [@appwrite](https://twitter.com/appwrite) on Twitter and/or email [team@appwrite.io](mailto:team@appwrite.io) so we can give pointers and tips and help you spread the word by promoting your content on the different Appwrite communication channels. Please add your blog posts and videos of talks to our [Awesome Appwrite](https://github.com/appwrite/awesome-appwrite) repo on GitHub.

### Presenting at Meetups

Presenting at meetups and conferences about your Appwrite projects. Your unique challenges and successes in building things with Appwrite can provide great speaking material. We’d love to review your talk abstract/CFP, so get in touch with us if you’d like some help!

### Sending Feedbacks & Reporting Bugs

Sending feedback is a great way for us to understand your different use cases of Appwrite better. If you had any issues, bugs, or want to share about your experience, feel free to do so on our GitHub issues page or at our [Discord channel](https://discord.gg/GSeTUeA).

### Submitting New Ideas

> If you think Appwrite could use a new feature, please open an issue on our GitHub repository, stating as much information as you can think about your new idea and it's implications. We would also use this issue to gather more information, get more feedback from the community, and have a proper discussion about the new feature.

Reviewer: ```suggestion
If you think Appwrite could use a new feature, please open an issue on our GitHub repository, stating as much information as you can think about your new idea and its implications. We would also use this issue to gather more information, get more feedback from the community, and properly discuss the new feature.
```

Author: Maybe we should delete this whole section since there's "Introducing New Features" up above?

Reviewer: Yeah let's do that <3

Author: updated!
